mozci/push.py

```diff
@@ -446,14 +464,40 @@ def get_candidate_regressions(self, runnable_type):
                 candidate_regressions[name] = (count, summary.status)
 
             other = other.child
-            # Any failure that comes after this push has been backed-out, can't
-            # be blamed on this push.
-            if self.backedoutby in other.revs:
-                break
             count += 1
 
         return candidate_regressions
 
+    @memoized_property
+    def bustage_fixed_by(self):
+        """The revision of the commit which 'bustage fixes' this one or None.
+
+        We detect if a push was 'bustage fixed' with a simple heuristic:
+        - there is a close enough child push where the task/group passes;
+        - the child push where the task/group passes is associated to the same bug
+          as the push of interest.
+
+        Returns:
+            str or None: The commit revision which 'bustage fixes' this push (or None).
+        """
+        if self.backedout:
+            return False
+
+        count = 0
+        other = self.child
+        while count < MAX_DEPTH:
+            if len(self.bugs & other.bugs) > 0 and any(
+                name in other.label_summaries
+                and other.label_summaries[name].status == Status.PASS
+                for name in self.get_candidate_regressions("label").keys()
```

Comment:
Looks like we walk down the childs both here and in `get_candidate_regressions`. Could we try to make sure we only walk down children once?

Maybe a `_process_children` helper that sets everything up the first time it is called and is a no-op on subsequent times. I guess the expensive parts are memoized anyway, so maybe it's not worth it.. Feel free to ignore this for now.

Reply:
It makes sense to do it at some point anyway, as we could use the `bustage_fixed_by` to check the sheriffs' classifications.

Reply:
To clarify my previous comment: if we don't untangle bustage_fixed_by from get_candidate_regressions, we would create a circular dependency.

I've filed #125.